build_keyboard.mk

```diff
@@ -290,6 +290,9 @@ ifneq ("$(wildcard $(USER_PATH)/config.h)","")
     CONFIG_H += $(USER_PATH)/config.h
 endif
 
+# Disable features that a keyboard doesn't support
+-include disable_features.mk
```

Comment:
This should probably be moved to before the userspace code here, so that the changes are applied before the userspace checks.

Reply:
I've been thinking about this but I don't see why. Can you expand on your thinking here?

Reply:
As mentioned earlier, this currently doesn't disable the feature before the userspace `rules.mk` file is processed. This is problematic because that file will see the feature as enabled, even though the keyboard has flagged it as not supported.

Consider the following code in a userspace `rules.mk` file:
```c
ifeq ($(strip $(BACKLIGHT_ENABLE)), yes)
    SRC += my_backlight.c
endif
```
If `BACKLIGHT_ENABLE = yes` is set in this user's community layout and/or userspace, `my_backlight.c` will be included and compiled even if the keyboard declares `BACKLIGHT_SUPPORTED = no`.

Drashna's suggestion of moving this line before the userspace code fixes this problem.

Reply:
Could that be solved by incorporating this into your logic?

```
ifneq ($(strip $(BACKLIGHT_SUPPORTED)), no)
```

Reply:
Yes, it works when the check is changed to:
```make
ifneq ($(strip $(BACKLIGHT_SUPPORTED)), no)
    ifeq ($(strip $(BACKLIGHT_ENABLE)), yes)
        SRC += my_backlight.c
    endif
endif
```
Is this something that you think would be worth doing for all similar checks moving forward? Or is the idea that `<FEATURE>_SUPPORTED` should automatically (and correctly) work with existing code that relies on just `<FEATURE>_ENABLED`?

Reply:
I'm not sure I'm ready to make that call yet, right now I'm gathering data. This does seem like a potential solution for at least some use cases.

Reply:
Please see this for an alternative potential solution: https://github.com/qmk/qmk_firmware/pull/9070#issuecomment-627679188.